service_manager/service_manager.py

```diff
@@ -0,0 +1,49 @@
+import asyncio
+from functools import partial
+
+from yapapi.log import enable_default_logger
+
+from .service_wrapper import ServiceWrapper
+from .yapapi_connector import YapapiConnector
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
```

Comment:
Interesting 🤔 ? What it means? I thought that annotating the types is the developer decision 😕

Reply:
`if TYPE_CHECKING` means "execute this code only when checking types in a static way (e.g. when running `mypy`)".
Afaik this is the standard python way of saying "we need those only to make `mypy` work". @Wiezzel understands this better I think :P

Reply:
https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING
This constant is `True` when the code is analyzed by a type-checker but `False` otherwise. This way we don't execute typing-related import when running the code.